core/math/matrix3.cpp

```diff
@@ -369,32 +369,24 @@ Vector3 Basis::get_euler_xyz() const {
 
 	Vector3 euler;
 #ifdef MATH_CHECKS
-	ERR_FAIL_COND_V(is_rotation() == false, euler);
+	ERR_FAIL_COND_V(!is_rotation(), euler);
 #endif
-	euler.y = Math::asin(elements[0][2]);
-	if (euler.y < Math_PI * 0.5) {
-		if (euler.y > -Math_PI * 0.5) {
-			// is this a pure Y rotation?
```

Comment:
This two ifs assume that Math::asin(1) == Math_PI * 0.5 and Math::asin(-1) == Math_PI * 0.5 which I'm pretty sure it is not true in floating point arithmetcs